contracts/Comptroller.sol

```diff
@@ -1019,6 +1030,40 @@ contract Comptroller is ComptrollerV3Storage, ComptrollerInterface, ComptrollerE
             require(allMarkets[i] != CToken(cToken), "market already added");
         }
         allMarkets.push(CToken(cToken));
+
+        // Initially, set borrow limit to unlimited. This should be set appropriately in the same governance call immediatly after adding the market.
+        _setMarketBorrowLimitInternal(CToken(cToken),uint(-1));
+
+    }
+
+
+    /**
+      * @notice Set the given borrow limit for the given cToken market
+      * @dev Admin function to set the borrow limit
+      * @param cToken The address of the market (token) to change the borrow limit for
+      * @param borrowLimit The new borrow limit value in underlying (maximum borrowing) to be set
+      * @return uint 0=success, otherwise a failure. (See enum Error for details)
+      */
+    function _setMarketBorrowLimit(CToken cToken, uint256 borrowLimit) public returns (uint) {
+        // Allow pause guardian to change borrow limit too.
+        if (msg.sender != admin && msg.sender != pauseGuardian) {
```

Comment:
Interesting that we give this power to the pause guardian. If we are doing that, we should probably follow the same policy as before and allow the pause guardian to _decrease_ the limit, but only the governor to _increase_ it.

Reply:
Put this is the details of the PR:

"I allow the pause guardian to change the borrow limit because in times of high network traffic, if a market quickly reaches the borrow limit, there may be issues for liquidators who borrow to repay. In order to ensure that liquidations can continue as normal, the borrow limit will need to be adjusted quickly."

I discussed this with a high volume compound liquidator and they were confident that many liquidation bots would break if we hit the borrow limit. 5 day delay could cause lots of bad debt.